torch/csrc/utils/variadic.h

```diff
@@ -108,4 +108,50 @@ struct MakeIndices : MakeIndices<N-1, N-1, Is...> {};
 // 0 through N-1.
 template<size_t... Is>
 struct MakeIndices<0, Is...> { using indices = Indices<Is...>; };
+
+//===----------------------------------------------------------------------===//
+//                                 Utilities
+//===----------------------------------------------------------------------===//
+
+template <bool value, typename T = void>
+using enable_if_t = typename std::enable_if<value, T>::type;
+
+template <bool value, typename T = void>
+using disable_if_t = enable_if_t<!value, T>;
+
+template <typename T>
+using decay_t = typename std::decay<T>::type;
+
+namespace detail {
+template <bool...>
+struct pack;
+} // namespace detail
+
+template <bool... values>
+struct all_of : std::is_same<
+                    detail::pack<values..., true>,
+                    detail::pack<true, values...>> {};
+
+template <bool...>
+struct any_of_t;
```

Comment:
Usually makes sense to define `any_of` first and then `any_of_t` based on it. Later in c++17, someone probably wants to add `any_of_v` based on the same implementation.